#! /bin/sh /usr/share/dpatch/dpatch-run
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Prevent gcc from optimizing memcpys away. Fixes SIGBUS on sparc
## DP: #485525

@DPATCH@
--- a/modules/ssl/ssl_scache_shmcb.c.dist	2008-07-18 17:11:20.839017290 +0000
+++ b/modules/ssl/ssl_scache_shmcb.c	2008-07-18 17:57:52.963016329 +0000
@@ -288,6 +288,9 @@
     return ret;
 }
 
+#if defined(__GNUC__) && __GNUC__ > 3
+__attribute__((__noinline__))
+#endif
 static void shmcb_set_safe_uint_ex(unsigned char *dest,
                                 const unsigned char *src)
 {
@@ -303,6 +306,9 @@
     return ret;
 }
 
+#if defined(__GNUC__) && __GNUC__ > 3
+__attribute__((__noinline__))
+#endif
 static void shmcb_set_safe_ulong_ex(unsigned char *dest,
                                 const unsigned char *src)
 {
@@ -318,6 +324,9 @@
     return ret;
 }
 
+#if defined(__GNUC__) && __GNUC__ > 3
+__attribute__((__noinline__))
+#endif
 static void shmcb_set_safe_time_ex(unsigned char *dest,
                                 const unsigned char *src)
 {
